grails-web-common/src/test/groovy/grails/web/servlet/mvc/GrailsParameterMapTests.groovy

```diff
@@ -294,27 +293,36 @@ class GrailsParameterMapTests extends GroovyTestCase {
         assertEquals true, map.boolean('my_checkbox')
     }
 
-    void testAutoEvaluateBlankDates() {
+    @Issue("https://github.com/grails/grails-core/issues/11126")
+    void testDontAutoEvaluateBlankDates() {
         mockRequest.addParameter("foo", "date.struct")
         mockRequest.addParameter("foo_year", "")
         mockRequest.addParameter("foo_month", "")
 
         theMap = new GrailsParameterMap(mockRequest)
-        assert theMap['foo'] == null : "should be null"
+        assert theMap['foo'] == "date.struct" : "should not be modified"
     }
 
-    void testAutoEvaluateDates() {
+    @Issue("https://github.com/grails/grails-core/issues/11126")
+    void testDontAutoEvaluateDates() {
         mockRequest.addParameter("foo", "date.struct")
         mockRequest.addParameter("foo_year", "2007")
         mockRequest.addParameter("foo_month", "07")
 
         theMap = new GrailsParameterMap(mockRequest)
+        assert theMap['foo'] == "date.struct" : "should not be modified"
+    }
 
-        assert theMap['foo'] instanceof Date : "Should have returned a date but was a ${theMap['foo']}!"
-        def cal = new GregorianCalendar()
-        cal.setTime(theMap['foo'])
+    @Issue("https://github.com/grails/grails-core/issues/11126")
+    void testGetDateDoesConversion() {
+        mockRequest.addParameter("foo", "date.struct")
+        mockRequest.addParameter("foo_year", "2007")
+        mockRequest.addParameter("foo_month", "07")
 
-        assert 2007 == cal.get(Calendar.YEAR) : "Year should be 2007"
+        theMap = new GrailsParameterMap(mockRequest)
+        assert theMap.getDate("foo").getClass() == java.util.Date
```

Comment:
@ilopmar this is no longer behaving like this in grails 4.0.0. Can this fix be ported to grails 4?